src/xform_to_datalog/datalog_facts_test.cc

```diff
@@ -0,0 +1,492 @@
+    absl::StrJoin({kGenRecipeName, kH2HandleName, kField1Name}, ".");
+
+static const std::string kGenRecipeH3Field1 =
+    absl::StrJoin({kGenRecipeName, kH3HandleName, kField1Name}, ".");
+
+static const std::string kGenRecipeH4Field1 =
+    absl::StrJoin({kGenRecipeName, kH4HandleName, kField1Name}, ".");
+
+static const std::string kGenRecipePs1Inst0InHandleField1 =
+    absl::StrJoin({kGenRecipeName, kPs1Inst0Name, kInHcName, kField1Name}, ".");
+
+static const std::string kGenRecipePs1Inst0InHandleField2 =
+    absl::StrJoin({kGenRecipeName, kPs1Inst0Name, kInHcName, kField2Name}, ".");
+static const std::string kGenRecipePs1Inst0OutHandleField1 =
+    absl::StrJoin({kGenRecipeName, kPs1Inst0Name, kOutHcName, kField1Name},
+                  ".");
+
+static const std::string kGenRecipePs2Inst1InHandleField1 =
+    absl::StrJoin({kGenRecipeName, kPs2Inst1Name, kInHcName, kField1Name}, ".");
+
+static const std::string kGenRecipePs2Inst1OutHandleField1 =
+    absl::StrJoin({kGenRecipeName, kPs2Inst1Name, kOutHcName, kField1Name},
+                  ".");
+
+static const std::string kGenRecipePs2Inst2InHandleField1 =
+    absl::StrJoin({kGenRecipeName, kPs2Inst2Name, kInHcName, kField1Name}, ".");
+
+static const std::string kGenRecipePs2Inst2OutHandleField1 =
+    absl::StrJoin({kGenRecipeName, kPs2Inst2Name, kOutHcName, kField1Name},
+                  ".");
+
+static std::string MakeClaimStr(std::string access_path_str, std::string tag) {
+  return absl::StrCat(
+      "claimHasTag(\"", access_path_str, "\", \"", tag, "\").\n");
+}
+
+static const std::string kExpectedClaimStrings[] = {
+    MakeClaimStr(kNamedRecipePs1Inst0OutHandleField1, "tag1"),
+    MakeClaimStr(kNamedRecipePs1Inst1OutHandleField1, "tag1"),
+    MakeClaimStr(kNamedRecipePs2Inst2OutHandleField1, "tag3"),
+    MakeClaimStr(kGenRecipePs1Inst0OutHandleField1, "tag1"),
+    MakeClaimStr(kGenRecipePs2Inst1OutHandleField1, "tag3"),
+    MakeClaimStr(kGenRecipePs2Inst2OutHandleField1, "tag3"),
+};
+
+static std::string MakeCheckStr(std::string access_path_str, std::string tag) {
+  return absl::StrCat(
+      "checkHasTag(\"", access_path_str, "\", \"", tag, "\").\n");
+}
+
+static std::string kExpectedCheckStrings[] = {
+    MakeCheckStr(kNamedRecipePs1Inst0InHandleField1, "tag2"),
+    MakeCheckStr(kNamedRecipePs1Inst1InHandleField1, "tag2"),
+    MakeCheckStr(kNamedRecipePs2Inst2InHandleField1, "tag4"),
+    MakeCheckStr(kGenRecipePs1Inst0InHandleField1, "tag2"),
+    MakeCheckStr(kGenRecipePs2Inst1InHandleField1, "tag4"),
+    MakeCheckStr(kGenRecipePs2Inst2InHandleField1, "tag4"),
+};
+
+static std::string MakeEdgeStr(std::string ap1, std::string ap2) {
+  return absl::StrCat("edge(\"", ap1, "\", \"", ap2, "\").\n");
+}
+
+static const std::string kExpectedEdgeStrings[] = {
+    // Named recipe edges:
+    // Edges connecting h1 to NamedR.PS1#0 for fields {field1, field2}.
+    MakeEdgeStr(kNamedRecipeH1Field1, kNamedRecipePs1Inst0InHandleField1),
```

Comment:
Having the strings explicitly written as raw strings would be much more readable, I think.

Reply:
Done.

Reply:
No changes here? Did you forget to push the changes to the branch?

In any case, I meant that instead of `MakeEdgeStr(...)` let us have the strings like `"edge(...)"` here.

Reply:
Sorry, I misunderstood what you meant. I thought you wanted to have the format string within the helper function be written as a raw string.

I went back and forth on this a lot. I wanted to prevent a situation in which a typo in a commonly-used component of these strings caused frustration in searching for the result. However, I think I may have made the wrong call: reading the expectations readily is probably more important. I'll switch as you suggest.

Reply:
Alright, I pushed a commit that should actually do what you asked for here. Sorry about the confusion!